File: android/app/src/main/java/com/iceblox/app/processing/PlateHasher.kt
package com.iceblox.app.processing

import com.iceblox.app.BuildConfig
import javax.crypto.Mac
import javax.crypto.spec.SecretKeySpec

object PlateHasher {
    private val pepperKey: ByteArray = BuildConfig.PEPPER.toByteArray(Charsets.UTF_8)

    fun hash(normalizedPlate: String): String {
        val mac = Mac.getInstance("HmacSHA256")
        mac.init(SecretKeySpec(pepperKey, "HmacSHA256"))
        val digest = mac.doFinal(normalizedPlate.toByteArray(Charsets.UTF_8))
        return digest.joinToString("") { "%02x".format(it) }
    }
}

File: ios/IceBloxApp/Processing/PlateHasher.swift
import CryptoKit
import Foundation

enum PlateHasher {
    private static var pepperKey: SymmetricKey = {
        SymmetricKey(data: Data(Pepper.value.utf8))
    }()

    static func hash(normalizedPlate: String) -> String {
        let data = Data(normalizedPlate.utf8)
        let mac = HMAC<SHA256>.authenticationCode(for: data, using: pepperKey)
        return mac.map { String(format: "%02x", $0) }.joined()
    }
}
